.github/workflows/releaseArtifacts.yml

```diff
@@ -14,6 +14,8 @@ jobs:
     permissions:
       contents: write
       packages: write
+      id-token: write
+      attestations: write
```

Comment:
Giving write permissions when building a package can lead to a supply chain attack. I would advise splitting the process into two separate jobs. The first job will build the package and upload the result to the artifact cache using [actions/upload-artifact](https://github.com/actions/upload-artifact). The second job, with the two write permissions, will download the artifact from the cache using [actions/download-artifact](https://github.com/actions/download-artifact), sign it, and upload both the signed artifact and its signature.